modules/Bio/EnsEMBL/Compara/Production/Analysis/AlignmentChains.pm

```diff
@@ -91,16 +78,13 @@ sub run_chains {
     $target_nib_dir =  "$work_dir/target_nib";
     mkdir $target_nib_dir;
 
-    my $target_slices = $self->param('target_slices');
-    foreach my $nm (keys %{$target_slices}) {
-      my $target = $target_slices->{$nm};
-      my $target_name = $target->seq_region_name;
-      
-      my $seqio =  Bio::SeqIO->new(-format => 'fasta',
-                                -file   => ">$target_nib_dir/$target_name.fa");
-      $seqio->write_seq($target);
-      $seqio->close; 
-      
+    my $target_chunks = $self->param('target_chunks');
+    foreach my $nm (keys %{$target_chunks}) {
+      my $target = $target_chunks->{$nm};
+      my $target_name = $target->dnafrag->name;
```

Comment:
Isn't `$target_name` the same as `$nm`? Why defined it twice? 

```suggestion
    foreach my $target_name (keys %{$target_chunks}) {
      my $target = $target_chunks->{$target_name};
```

Reply:
I have tried to leave as much of the original code as possible, but you are right that before these two weren't the same and now they are. Well spotted!